core/backend/src/CustomViewState3dCreator.ts

```diff
@@ -59,22 +59,29 @@ export class CustomViewState3dCreator {
     if (modelIdsList.length === 0)
       return modelExtents;
     const modelIds = new Set(modelIdsList);
-    Logger.logInfo(loggerCategory, "Starting getModelExtents query.");
     for (const id of modelIds) {
+      const modelExtentsStopWatch = new StopWatch("getModelExtents query", false);
       try {
         await new Promise((resolve) => setImmediate(resolve)); // Free up main thread temporarily. Ideally we get queryModelExtents off the main thread and do not need to do this.
+        Logger.logInfo(loggerCategory, "Starting getModelExtents query.", () => ({modelId: id}));
+        modelExtentsStopWatch.start();
         const props = this._imodel.nativeDb.queryModelExtents({ id }).modelExtents;
+        modelExtentsStopWatch.stop();
+        Logger.logInfo(loggerCategory, "Finished getModelExtents query.", () => ({timeElapsedMs: modelExtentsStopWatch.elapsed, modelId: id}));
         modelExtents.union(Range3d.fromJSON(props), modelExtents);
       } catch (err: any) {
+        modelExtentsStopWatch.stop();
         if ((err as IModelError).errorNumber === IModelStatus.NoGeometry) { // if there was no geometry, just return null range
+          Logger.logInfo(loggerCategory, "Finished getModelExtents query with NoGeometry error.", () => ({timeElapsedMs: modelExtentsStopWatch.elapsed, modelId: id}));
```

Comment:
there is no reason to make this an arrow function. Just pass the object.

Reply:
Oops, I just copied it from somewhere else in the repo. Changing it now